store/postgres/src/entities.rs

```diff
@@ -258,6 +370,18 @@ impl<'a> Connection<'a> {
         }
     }
 
+    /// Do any cleanup to bring the subgraph into a known good state
+    pub(crate) fn start_subgraph(&self, subgraph: &SubgraphDeploymentId) -> Result<(), StoreError> {
+        use public::deployment_schemas as dsl;
+
+        Ok(
+            diesel::update(dsl::table.filter(dsl::subgraph.eq(subgraph.to_string())))
+                .set(dsl::migrating.eq(false))
```

Comment:
No, `migrating` is set to true in its own transaction, the migration work happens in separate txns. That's necessary because `migrating` is essentially a lock, and taking that lock needs to be visible to all db users _before_ the migration work starts. And if the server crashes/gets killed, we are left with `migrating` set to true without actually doing migration work.

Reply:
Makes sense, thanks for the explanation. Maybe you can add that example of a server crash as a comment on this line?

Reply:
Yep, good idea. Just did that.